atlp-rwanda/stackup2-barefoot-backend
# #171380734 Configure ESLint

#### What does this PR do?
Adds ESLint as a style guide linter to the project
#### Description of Task to be completed?
- create a .eslintrc.json file to hold configurations and AirBnB style guide rules
- Install `eslint ` npm package for linting
- Install `eslint-config-airbnb` npm package for AirBnB base configurations(Some rules are overridden in eslintrc.json file)
- Install `eslint-plugin-import` npm package to support linting of ES2015+ (ES6+) import/export syntax, and prevent issues with misspelling of file paths and import names
#### How should this be manually tested?
- After cloning the project repo, run `npm i` inside the cloned repo folder
- run `git pull <remote name>(e.g:origin) ch-eslint-configuration-171380734`
- If you are using vs-code text editor, [install ESLint plugin for vs-code](https://marketplace.visualstudio.com/items?itemName=dbaeumer.vscode-eslint)
- configure eslint as the default formatter for your editor 
- create a JavaScript file inside the project folder you just cloned(preferably src folder) and add some JavaScript code, you can use `var invalid_var  =  0;` for testing
- If all of the above steps have been completed properly, vs-code should display some error messages
- In case there are some errors, run `./node_modules/.bin/eslint --fix <path to file||folder with errors>` to fix them
#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
#171380734
#### Screenshots (if appropriate)
N/A
